benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/scheduler/TestTaskScheduler.java

```diff
@@ -189,7 +190,8 @@ public synchronized void handleRunningTest(String testID) {
         }
         // Exit while waiting for the Experiment Manager to notify about the scheduled
         // experiment to be executed. This exits before the execution of HANDLE_EXPERIMENT_RESULT
-        else if (testRunningState != null && testRunningState.equals(HANDLE_EXPERIMENT_RESULT)) {
+        else if (testRunningState != null
+            && testRunningState.equals(TestRunningState.HANDLE_EXPERIMENT_RESULT)) {
```

Comment:
I think also here we need to check for `|| testRunningState == TestRunningState.TERMINATING`